From a6931d3756d991cab1904a91a2fd2a094da8b2f2 Mon Sep 17 00:00:00 2001
From: Mark Cox <mjc@redhat.com>
Date: Thu, 2 Jan 2014 13:13:08 +0000
Subject: Use mktemp in groffer to eliminate a race condition

 This was forwarded in
 http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=278265#12 and addressed
 differently, but it would still be much better to use mktemp if available.
Origin: vendor, https://bugzilla.redhat.com/show_bug.cgi?id=136313
Author: Colin Watson <cjwatson@debian.org>
Bug-Debian: http://bugs.debian.org/278265
Forwarded: no
Last-Update: 2010-03-01

Patch-Name: groffer-temporary-files.patch
---
 contrib/groffer/shell/groffer2.sh | 23 +----------------------
 1 file changed, 1 insertion(+), 22 deletions(-)

diff --git a/contrib/groffer/shell/groffer2.sh b/contrib/groffer/shell/groffer2.sh
index d76c6bd..be632d5 100644
--- a/contrib/groffer/shell/groffer2.sh
+++ b/contrib/groffer/shell/groffer2.sh
@@ -5222,28 +5222,7 @@ main_init()
       _TMP_DIR="${mi_dir}"'/';
       ;;
     esac;
-    _TMP_DIR="${_TMP_DIR}groffer${_PROCESS_ID}";
-    if obj _TMP_DIR rm_tree
-    then
-      :
-    else
-      mi_tdir_="${_TMP_DIR}"_;
-      mi_n=1;
-      mi_tdir_n="${mi_tdir_}${mi_n}";
-### main_init()
-      while obj mi_tdir_n is_existing
-      do
-        if obj mi_tdir_n rm_tree
-        then
-          # directory could not be removed
-          mi_n="$(expr "${mi_n}" + 1)";
-          mi_tdir_n="${mi_tdir_}${mi_n}";
-          continue;
-        fi;
-      done;
-      _TMP_DIR="${mi_tdir_n}";
-    fi;
-    eval mkdir "${_TMP_DIR}";
+    _TMP_DIR="$(mktemp -d "${_TMP_DIR}groffer.XXXXXX")";
     if is_not_equal "$?" 0
     then
       obj _TMP_DIR rm_tree;
